hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/procedure/TestUpgradeOrDowngradeProcedure.scala

```diff
@@ -142,6 +143,56 @@ class TestUpgradeOrDowngradeProcedure extends HoodieSparkProcedureTestBase {
     }
   }
 
+  test("Test downgrade table from version eight to version seven") {
+    withTempDir { tmp =>
+      val tableName = generateTableName
+      val tablePath = s"${tmp.getCanonicalPath}/$tableName"
+      // create table
+      spark.sql(
+        s"""
+           |create table $tableName (
+           |  id int,
+           |  name string,
+           |  price double,
+           |  ts long
+           |) using hudi
+           | location '$tablePath'
+           | options (
+           |  type = 'mor',
+           |  primaryKey = 'id',
+           |  preCombineField = 'ts'
+           | )
+       """.stripMargin)
+
+      spark.sql("set hoodie.compact.inline=true")
+      spark.sql("set hoodie.compact.inline.max.delta.commits=1")
+      spark.sql("set hoodie.clean.commits.retained = 2")
+      spark.sql("set hoodie.keep.min.commits = 3")
+      spark.sql("set hoodie.keep.min.commits = 4")
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+      spark.sql(s"insert into $tableName values(1, 'a1', 10, 1000)")
+
+      var metaClient = createMetaClient(spark, tablePath)
+      // verify hoodie.table.version of the table is EIGHT
+      if (metaClient.getTableConfig.getTableVersion.versionCode().equals(HoodieTableVersion.EIGHT.versionCode())) {
+        // downgrade table from version eight to version seven
+        checkAnswer(s"""call downgrade_table(table => '$tableName', to_version => 'SEVEN')""")(Seq(true))
+        metaClient = HoodieTableMetaClient.reload(metaClient)
+        assertResult(HoodieTableVersion.SEVEN.versionCode) {
+          metaClient.getTableConfig.getTableVersion.versionCode()
+        }
+        // Verify whether the naming format of instant files is consistent with 0.x
+        metaClient.reloadActiveTimeline().getInstants.iterator().asScala.forall(f => !f.getFileName.contains("_"))
```

Comment:
Can we add a pattern in `HoodieInstant` and make this check more strict? For example, in HoodieInstant, we have `NAME_FORMAT` pattern which checks for both 0.x and 1.x name pattern. We can add a more strict pattern like NAME_FORMAT_0_X which should fail for 1.x instant file names.